src/pudl/etl.py

```diff
@@ -446,6 +446,76 @@ def _read_static_pudl_tables() -> dict[str, pd.DataFrame]:
     return {"political_subdivisions": POLITICAL_SUBDIVISIONS}
 
 
+def make_datasources_table(datasets_settings, ds):
+    """Compile a table of dataset information.
+
+    There are three places we can look for information about a dataset:
+     * the datastore (for DOIs, working partitions, etc)
+     * the ETL settings (for partitions that are used in the ETL)
+     * the DataSource info (which is stored within the ETL settings)
+
+    The ETL settings and the datastore have different levels of nesting - and therefor
+    names for datasets. The nesting happens particularly with the EIA data. There
+    are three EIA datasets right now - eia923, eia860 and eia860m. eia860m is a monthly
+    update of a few tables in the larger eia860 dataset.
+    """
```

Comment:
Can you add argument descriptions and type hints here, please?

Reply:
ah yes! duh.